src/v/cluster/seq_stm.cc

```diff
@@ -210,21 +78,23 @@ static bool is_sequence(int32_t last_seq, int32_t next_seq) {
 ss::future<checked<raft::replicate_result, kafka::error_code>>
 seq_stm::replicate(
   model::batch_identity bid,
-  model::record_batch_reader&& r,
+  model::record_batch_reader&& b,
   raft::replicate_options opts) {
     if (bid.has_idempotent() && bid.first_seq <= bid.last_seq) {
-        if (_insync_term != _c->term()) {
-            (void)ss::with_gate(_gate, [this] { return catchup(); });
-            return seastar::make_ready_future<
-              checked<raft::replicate_result, kafka::error_code>>(
+        // it's very wierd but moving the r-value from the arg
+        // to stack fixes a seg fault
+        auto br = std::move(b);
+        auto timeout = config::shard_local_cfg().seq_sync_timeout_ms.value();
+        auto is_ready = co_await sync(timeout);
+        if (!is_ready) {
+            co_return checked<raft::replicate_result, kafka::error_code>(
```

Comment:
Ok, I'm convinced, but I don't want to change scope of this PR so I've created a task to follow up: https://github.com/vectorizedio/redpanda/issues/841

Reply:
sounds good. thanks for create the ticket :) this is also related to the issue with the schema generator running and the build error where one of the schema files was not created because cluster doesn't have a dependency on kafka. so it might come up again.